### PR TITLE
feat(superchain): enable local testing

### PR DESCRIPTION
**Pull request by edobry**
### Description

This PR modifies the `superchain` configuration initialization logic to replace the `PublicRPC` endpoints not only in CI but also when running validation tests locally, as many of them require access to an L1 archive node in order to function properly. The previously-hardcoded URLs are extracted to `test_rpc` config values next to the existing `public_rpc` values they override.

This change required a modification to the `proxyd-l1-ci` IP whitelists to whitelist internal Tailnet IPs.

### Additional Context

See a discussion of the issue [here](https://oplabs-pbc.slack.com/archives/C02MWK11M27/p1733447920092209).

The [initial PR](https://github.com/ethereum-optimism/k8s/pull/5202) whitelisting Tailnet IPs was misconfigured, but then fixed in this [followup PR](https://github.com/ethereum-optimism/k8s/pull/5223).

### CODE REVIEW

**Review by bitwiseguy — CHANGES_REQUESTED** [review]
I think this could be a great improvement, but as it stands its not obvious why failures occur if trying to run validations locally without being connected to tailscale exit nodes. Forcing external users to "read the docs" to troubleshoot validations that previously worked is likely to cause friction. So I think it hurts the devx right now. 

If we're going to merge this work, I think we should lift the L1 client connection so that we only attempt that once, instead of inside each validation. If the connection is successful, we pass that client connection to each underlying validation that needs it. If the connection fails (due to no tailscale connection), we should give a clear error message upfront (without running the full validation suite) about what to try (i.e. replace the rpcUrl in the relevant areas, and the url must be to an L1 archive node), and/or default/fallback to the current public full node urls

File: validation/data-availability-type_test.go [hunk 1]
```diff
@@ -20,10 +20,7 @@ import (
 
 func testDataAvailabilityType(t *testing.T, chain *ChainConfig) {
 	rpcEndpoint := Superchains[chain.Superchain].Config.L1.PublicRPC
-	require.NotEmpty(t, rpcEndpoint)
-
-	client, err := ethclient.Dial(rpcEndpoint)
-	require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
```
[the remark this answers is not in the document]
Comment (edobry) [reply validation/data-availability-type_test.go:26]: in theory since this condition was already checked at the start of the test suite run, it shouldn't fail here either besides in edge-case network conditions. do you think that warrants keeping them in? i'm fine reverting if so, just didn't see much value in that given the small time difference between them

[the remark this answers is not in the document]
Comment (bitwiseguy) [reply validation/data-availability-type_test.go:26]: Yea I think if we want to eliminate these redundant connection error checks, we should just make a single ethclient connection and pass the client to each underlying dependent. We are working towards that in some of the refactors discussed [here](https://github.com/ethereum-optimism/superchain-registry/pull/802#discussion_r1889403017). I'd say revert the error check removal for now and we'll get rid of the redundancy there.

**Comment by mslipper** [comment]
These files are removed in SR2.0, closing.
